base/loading.jl

```diff
@@ -1204,53 +1203,56 @@ function run_extension_callbacks(extid::ExtensionId)
     return succeeded
 end
 
-function run_extension_callbacks(pkgid::PkgId)
+function run_extension_callbacks()
     assert_havelock(require_lock)
-    # take ownership of extids that depend on this pkgid
-    extids = pop!(EXT_DORMITORY, pkgid, nothing)
-    extids === nothing && return
-    for extid in extids
-        if extid.ntriggers > 0
-            # It is possible that pkgid was loaded in an environment
-            # below the one of the parent. This will cause a load failure when the
-            # pkg ext tries to load the triggers. Therefore, check this first
-            # before loading the pkg ext.
-            pkgenv = Base.identify_package_env(extid.id, pkgid.name)
-            ext_not_allowed_load = false
-            if pkgenv === nothing
-                ext_not_allowed_load = true
-            else
-                pkg, env = pkgenv
-                path = Base.locate_package(pkg, env)
-                if path === nothing
+    for pkgid in copy(keys(EXT_DORMITORY))
+        haskey(Base.loaded_modules, pkgid) || continue
```

Comment:
would this be any better?
```suggestion
    for pkgid in intersect(keys(Base.loaded_modules), keys(Base.EXT_DORMITORY))
```

I also wonder if there is any meaningful way to sort this list to be predictable (if that is worthwhile to do), for example we could try to sort the result, but `isless(::Base.PkgId, ::Base.PkgId)` is not defined.

Reply:
I did this and also sorted by UUID.

Reply:
I hope folks don't decide to give their packages low or high uuids to hack trigger order.

Reply:
Doesn't really matter, this is just to keep it consistent.